code/game/machinery/engi_points_vendor.dm

```diff
@@ -0,0 +1,197 @@
+
+/obj/machinery/engi_points_manager
+	name = "Engineering Points Manager"
+	desc = "Who's a good boy?"
+	icon = 'icons/obj/machines/engi_points.dmi'
+	icon_state = "store"
+	verb_say = "states"
+	density = 1
+	anchored = 1
+	req_access = list(access_engine)
+	var/obj/item/device/radio/radio
+	var/GBP = 0
+	var/GBPearned = 0
+	var/power_export_bonus = 0
+	var/air_alarm_bonus = 0
+	var/power_alarm_bonus = 0
+	var/fire_alarm_bonus = 0
+	var/alarm_rating = ""
+	var/prior_bonus = 2500
+	var/total_bonus = 0
+	var/GBP_alarm_cooldown = 4500
+	var/list/prize_list = list()
+
+/datum/GBP_equipment
+	var/equipment_name = "generic"
+	var/equipment_path = null
+	var/cost = 0
+	var/amount = 0
+
+/datum/GBP_equipment/New(name, path, cost, amount)
+	equipment_name = name
+	equipment_path = path
+	cost = cost
+	amount = amount
+
+/obj/machinery/engi_points_manager/Initialize()
+	engi_points_list += src
+	radio = new(src)
+	radio.listening = 0
+	radio.frequency = 1357
+	prize_list = list(

```

Comment:
Cyberboss, but I put off his request until someone else seconded it in IRC and here we are